src/rebuild/srv.c

```diff
@@ -1063,16 +1045,37 @@ rebuild_leader_start(struct ds_pool *pool, uint32_t rebuild_ver,
 		D_GOTO(out, rc);
 	}
 
-	/* broadcast scan RPC to all targets */
-	rc = rebuild_scan_broadcast(pool, *p_rgt, tgts_failed, svc_list,
-				    map_ver, &map_buf_iov);
+	/* IV bcast the pool map in case for offline rebuild */
+	rc = pool_iv_map_update(pool, map_buf_iov.iov_buf, map_ver);
+	if (rc) {
+		D_ERROR("pool_iv_map_update failed %d.\n", rc);
+		D_GOTO(out, rc);
+	}
 	D_FREE(map_buf_iov.iov_buf);
+
+	rc = ds_pool_prop_fetch(pool, DAOS_PO_QUERY_PROP_ALL, &prop);
+	if (rc) {
+		D_ERROR("pool prop fetch failed: rc %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	/* Update pool properties by IV */
+	rc = pool_iv_prop_update(pool, prop);
```

Comment:
seems with possible race with pool_iv_prop_update() -> pool_iv_prop_update().
is the purpose only to propagate svc_list? if that is true seems the it is simpler and better that just bcast svc_list in rebuild_scan_bcast as current master code then need not the extra prop_update above.

Reply:
what is the race you mean here? Note: only stream 0 will be able to access iv in DAOS.

Yes, the purpose is to propagate svc_list to all rebuild servers, but we will move the rebuild initiator part to a separate/common module to iosrv, carrying this by rebuild RPC will cause a lot trouble, and also break the layer, so svc_list propagation will be done through IV.

Reply:
for the race I mean for example user update pool prop between the read prop and update prop here, as what we discussed maybe fine to only update svc list through IV here, or adding IV version.